File: src/main/java/ar/edu/um/haberes/rest/model/view/CodigoPeriodo.java
/**
 * 
 */
package ar.edu.um.haberes.rest.model.view;

import java.io.Serializable;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.IdClass;
import jakarta.persistence.Table;

import org.hibernate.annotations.Immutable;
import ar.edu.um.haberes.rest.model.view.pk.CodigoPeriodoPK;
import lombok.Data;

/**
 * @author daniel
 *
 */
@Data
@Entity
@Immutable
@Table(name = "vw_codigo_periodo")
@IdClass(CodigoPeriodoPK.class)
public class CodigoPeriodo implements Serializable {
	/**
	* 
	*/
	private static final long serialVersionUID = 4612018973560089942L;

	@Id
	private Integer anho;

	@Id
	private Integer mes;

	@Id
	private Integer codigoId;
}

File: src/main/java/um/haberes/rest/kotlin/model/view/CodigoPeriodo.kt
package um.haberes.rest.kotlin.model.view

import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import jakarta.persistence.UniqueConstraint
import org.hibernate.annotations.Immutable

@Entity
@Table(name = "vw_codigo_periodo",
    uniqueConstraints = [UniqueConstraint(columnNames = ["anho", "mes", "codigoId"])])
@Immutable
data class CodigoPeriodo(

    @Id
    var uniqueId: String? = null,

    var anho: Int? = null,
    var mes: Int? = null,
    var codigoId: Int? = null

)
